apache/superset
# chore(generator-superset): migrate to monorepo

### SUMMARY
Currently the `generator-superset` Yeoman generator template generates plugins that require the old `superset-ui` monorepo. This is a first-pass at fixing the generator by changing the following:
- Makes the plugin standalone by adding scripts for building the package, along with necessary babel and typescript configs
- Removes the option to create a `superset-ui` package - now the generator only supports creating plugins
- Remove the option to create class based components - now only functional components are provided out of the box
- Removes the option to add badges to the README file

### SCREENSHOT
This plugin has been generated with the updated template:
![image](https://user-images.githubusercontent.com/33317356/146784078-79937045-3a5d-4dfc-95fc-46f05cb1d382.png)

### TESTING INSTRUCTIONS
1. Go into the `superset-frontend/packages/generator-superset` folder and run `npm i`
2. Go into `~/src` and run `mkdir superset-plugin-chart-hello-world`
3. Run `yo @superset-ui/superset` and answer all questions
4. Run `npm i --force && npm run build && npm run test && npm pack` and see that the new package works and can be packaged for npm
5. Add the plugin to the Superset `MainPreset.js` and see it work

### ADDITIONAL INFORMATION
